recipe/conda_build_config.yaml

```diff
@@ -14,9 +14,7 @@ c_stdlib:
 m2w64_c_stdlib:                # [win]
   - m2w64-toolchain            # [win]
 c_stdlib_version:              # [unix]
-  - 2.12                       # [linux64 and os.environ.get("DEFAULT_LINUX_VERSION", "cos6") == "cos6"]
-  - 2.17                       # [linux64 and os.environ.get("DEFAULT_LINUX_VERSION", "cos6") == "cos7"]
```

Comment:
Removing these lines will break the ability to set the linux version in smithy via `os_version`

https://github.com/conda-forge/conda-smithy/blob/9e4720fc497f45f45d7bd319a354c762992c530d/conda_smithy/configure_feedstock.py#L994

Maybe we keep them and set the default to cos7?

Reply:
You mean you want to keep supporting `cos6` in some feedstocks?

Reply:
> You mean you want to keep supporting cos6 in some feedstocks?

Right. there are two things. First yes, we no longer officially support cos6. 

Second, are we deprecating `os_version` in smithy? If not, we need to keep the selectors here to account for smithy setting `DEFAULT_LINUX_VERSION` like this:

```
        # set the environment variable for OS version
        if platform == "linux":
            ver = forge_config["os_version"][f"{platform}_{arch}"]
            if ver:
                os.environ["DEFAULT_LINUX_VERSION"] = ver
```

We probably need to add them to add support for cos8 or higher, but in another PR maybe.

Reply:
> Second, are we deprecating os_version in smithy? If not, we need to keep the selectors here to account for smithy setting DEFAULT_LINUX_VERSION like this:

We should. As I mentioned for cos8, we should change `cdt_name` to be `conda` and apply that for `cos7` as well. Then we get rid of `DEFAULT_LINUX_VERSION`.

Reply:
Great. Then lgtm on this and we can move the deprecation discussion to smithy.